### PR TITLE
use_ws=True with all other system() classes

### PR DESCRIPTION
**Pull request by stepan-tsirkin**
allowed to have use_ws=True with all other system() classes

for that reason 

the property System.mp_grid  may be entered manually (and will override that from chk file in case of System_w90) if it is not provided inside the model.  

I added a test that compares results from chk with results from _tb.dat + use_ws . They match for quantities not involving AA_R. For AA_R there is some mismatch, it needs further investigation.

### CODE REVIEW

**Comment by jaemolihm** [comment]
What is the benefit of allowing `use_ws=True` for all systems? Is the goal just to make `use_ws=True` work for systems generated with wannier90 `_tb.dat` data? Or is the goal more general?

**Comment by stepan-tsirkin** [comment]
@jaemolihm , second main goal is to enable it for ASE, which will be my next PR.

Also, one can import a wannier90_hr file into PythTB, and then to wberri, although it is a very weird way, but @Liu-Xiaoxiong  did it for some reason.

Also it nay be useful for testing, comparing different interfaces.

**Comment by stepan-tsirkin** [comment]
So, any comments to this PR? shall I merge or do you want more time to look at it?  @jaemolihm  @Liu-Xiaoxiong

**Comment by jaemolihm** [comment]
I dont have time to look at it soon. @Liu-Xiaoxiong can you do the review?
In any case, I think you can just merge it @stepan-tsirkin because you covered all changes by the tests.

If you merge, please make an issue about the AAR issue that you mentioned because its a bug, (preferably with a script to reproduce it).

File: wannierberri/__system.py [hunk 1]
```diff
@@ -66,6 +70,13 @@ class System():
         set ``True`` if quantities derived from Qiao's approximated spin-current elements will be used. (QZYZ 2018). Default: ``{SHCqiao}``
     use_ws : bool
         minimal distance replica selection method :ref:`sec-replica`.  equivalent of ``use_ws_distance`` in Wannier90. Default: ``{use_ws}``
+    mp_grid : [nk1,nk2,nk3]
+        size of Monkhorst-Pak frid used in ab initio calculation. Needed when `use_ws=True`, and only if it cannot be read from input file, i.e.
```
Comment (jaemolihm) [inline wannierberri/__system.py:74]: Pack

File: wannierberri/__system.py [hunk 2]
```diff
@@ -66,6 +70,13 @@ class System():
...
+        like :class:`~wannierberri.System_tb`, :class:`~wannierberri.System_PythTB`, :class:`~wannierberri.System_PythTB` ,:class:`~wannierberri.System_fplo`, but only if
```
Comment (jaemolihm) [inline wannierberri/__system.py:75]: Pythtb occurs twice

File: wannierberri/__system.py [hunk 3]
```diff
@@ -66,6 +70,13 @@ class System():
         set ``True`` if quantities derived from Qiao's approximated spin-current elements will be used. (QZYZ 2018). Default: ``{SHCqiao}``
     use_ws : bool
         minimal distance replica selection method :ref:`sec-replica`.  equivalent of ``use_ws_distance`` in Wannier90. Default: ``{use_ws}``
+    mp_grid : [nk1,nk2,nk3]
+        size of Monkhorst-Pak frid used in ab initio calculation. Needed when `use_ws=True`, and only if it cannot be read from input file, i.e.
+        like :class:`~wannierberri.System_tb`, :class:`~wannierberri.System_PythTB`, :class:`~wannierberri.System_PythTB` ,:class:`~wannierberri.System_fplo`, but only if 
+        the data originate from ab initio data, not from toy models.
+        In contrast, for :class:`~wannierberri.System_w90` and :class:`~wannierberri.System_ase` it is not needed,  but can be provided and will override the original bvalue
```
Comment (jaemolihm) [inline wannierberri/__system.py:77]: Bvalue

**Review by jaemolihm — COMMENTED** [review]
Just some typos

File: wannierberri/__system.py [hunk 4]
```diff
@@ -151,11 +167,33 @@ def getSHC(self):
...
+            print ("using ws_distance")
+            ws_map=ws_dist_map(self.iRvec,self.wannier_centers_cart_ws, self.mp_grid,self.real_lattice, npar=self.npar)
+            for X in ['Ham','AA','BB','CC','SS','FF','SA','SHA','SR','SH','SHR']:
+                XR=X+'_R'
+                if hasattr(self,XR) :
+                    print ("using ws_dist for {}".format(XR))
+                    vars(self)[XR]=ws_map(vars(self)[XR])
+            self.iRvec=np.array(ws_map._iRvec_ordered,dtype=int)
+        else:
+            print ("NOT using wws_dist",self.use_ws,self.use_ws)
```
Comment (jaemolihm) [inline wannierberri/__system.py:187]: wws -> ws?

**Pull request merged**